### PR TITLE
[WIP] Add NameList parameters

### PR DESCRIPTION
**Pull request by nefrathenrici**
## Purpose 

Add NameList parameters from EDMF

## To-do

@trontrytel @haakon-e @costachris Please look over and edit the parameter names for clarity.
@nefrathenrici Figure out some way of documenting the NameList parameters for the future.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by trontrytel** [comment]
Thanks @nefrathenrici 

What is the format to use to add a longer description string that would contain  paper reference or some description of what the parameter does?

**Comment by trontrytel** [comment]
> Thanks @nefrathenrici
> 
> What is the format to use to add a longer description string that would contain paper reference or some description of what the parameter does?

For example:

```
[EDMF_max_area]
alias = "max_area"
value = 0.9
type = "float"
explanation = "Maximum allowed area of edmf updraft, See paper doi_link_to_amazing_paper.html, eq(6)"
```

**Comment by nefrathenrici** [comment]
@trontrytel This kind of docstring field would also be useful for autogenerating parameter documentation. 

@odunbar Do you have any preferences for this?

File: src/parameters.toml [hunk 1]
```diff
@@ -763,3 +763,211 @@ alias = "avogad"
...
+
+[EDMF_surface_area]
+alias = "surface_area"
+value = 0.1
+type = "float"
+
+[EDMF_max_area]
+alias = "max_area"
+value = 0.9
+type = "float"
```
Comment (haakon-e) [inline src/parameters.toml:778]:
```suggestion
[EDMF_max_area]
alias = "max_area"
value = 0.9
type = "float"
explanation = "Maximum area fraction per updraft. Parameter not described in the literature."
```

File: src/parameters.toml [hunk 2]
```diff
@@ -763,3 +763,211 @@ alias = "avogad"
...
+
+[EDMF_min_area]
+alias = "min_area"
+value = 1.0e-5
+type = "float"
```
Comment (haakon-e) [inline src/parameters.toml:783]:
```suggestion
[EDMF_min_area]
alias = "min_area"
value = 1.0e-5
type = "float"
explanation = "Minimum area fraction per updraft. Parameter not described in the literature."
```
```suggestion
[EDMF_min_area]
alias = "min_area"
value = 1.0e-5
type = "float"
```

File: src/parameters.toml [hunk 3]
```diff
@@ -763,3 +763,211 @@ alias = "avogad"
...
+
+[mixing_length_static_stab_coeff]
+alias = "static_stab_coeff"
+value = 0.4
+type = "float"
+
+[mixing_length_tke_surf_scale]
+alias = "tke_surf_scale"
+value = 3.75
+type = "float"
```
Comment (haakon-e) [inline src/parameters.toml:803]:
```suggestion
[mixing_length_tke_surf_scale]
alias = "tke_surf_scale"
value = 3.75
type = "float"
explanation = "Ratio of turbulence kinetic energy to squared friction velocity in the surface layer for the EDMF mixing length closure; denoted κ_*². See Lopez-Gomez et al. (2020) [https://doi.org/10.1029/2020MS002162], Table 1. Note: the square root, i.e. κ_*, is listed in the reference."
```

File: src/parameters.toml [hunk 4]
```diff
@@ -763,3 +763,211 @@ alias = "avogad"
...
+
+[mixing_length_Prandtl_number_scale]
+alias = "Prandtl_number_scale"
+value = 4.076923076923077
+type = "float"
+
+[mixing_length_Prandtl_number_0]
+alias = "Prandtl_number_0"
+value = 0.74
+type = "float"
```
Comment (haakon-e) [inline src/parameters.toml:813]:
```suggestion
[mixing_length_Prandtl_number_0]
alias = "Prandtl_number_0"
value = 0.74
type = "float"
explanation = "Turbulent Prandtl number in neutral conditions; denoted Pr_{t,0}. See Lopez-Gomez et al. (2020) [https://doi.org/10.1029/2020MS002162], Table 1 and Eq 36."
```

File: src/parameters.toml [hunk 5]
```diff
@@ -763,3 +763,211 @@ alias = "avogad"
...
+
+[mixing_length_Ri_crit]
+alias = "Ri_crit"
+value = 0.25
+type = "float"
```
Comment (haakon-e) [inline src/parameters.toml:818]:
```suggestion
[mixing_length_Ri_crit]
alias = "Ri_crit"
value = 0.25
type = "float"
explanation = "Critical gradient Richardson number. It is an upper limit to the gradient Richardson number . See Li (2019) [https://doi.org/10.1016/j.atmosres.2018.09.015], Section 6.2 for details"
```

**Review by haakon-e — COMMENTED** [review]
to be continued ...

File: src/parameters.toml [hunk 6]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[microphysics_prescribed_precipitation_fraction]
+alias = "prescribed_precip_frac"
+value = 1.0
+type = "float"
+
+[microphysics_precipitation_fraction_limiter]
+alias = "precip_fraction_limiter"
+value = 0.3
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:983]:
```suggestion
type = "float"
explanation = "Minimum precipitation fraction allowed. Should be between 0 and 1. Valid for simulations running with the 1-moment microphysics scheme, edmf turbulence convection model and cloud_cover precip_fraction_model."
```

File: src/parameters.toml [hunk 7]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[microphysics_model_precipitation_fraction]
+alias = "precip_fraction_model"
+value = "prescribed"
+type = "string"
+
+[microphysics_prescribed_precipitation_fraction]
+alias = "prescribed_precip_frac"
+value = 1.0
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:978]:
```suggestion
type = "float"
explanation = "Prescribed value of precipitation fraction. Should be between 0 and 1. Valid for simulations running with the 1-moment microphysics scheme, edmf turbulence convection model and prescribed precip_fraction_model. Not documented in the literature (yet)."
```

File: src/parameters.toml [hunk 8]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_closure_drag]
+alias = "pressure_closure_drag"
+value = "normalmode"
+type = "string"
+
+[microphysics_model_precipitation_fraction]
+alias = "precip_fraction_model"
+value = "prescribed"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:973]: explanation = "Precipitation fraction model choice. Options are prescribed (default) and cloud_cover. Valid for simulations running with the 1-moment microphysics scheme, edmf turbulence convection model. Not documented in the literature (yet)."

Comment (trontrytel) [inline src/parameters.toml:973]:
```suggestion
type = "string"
explanation = "Precipitation fraction model choice. Options are prescribed (default) and cloud_cover. Valid for simulations running with the 1-moment microphysics scheme, edmf turbulence convection model. Not documented in the literature (yet)."
```

File: src/parameters.toml [hunk 9]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_closure_buoy]
+alias = "pressure_closure_buoy"
+value = "normalmode"
+type = "string"
+
+[pressure_closure_drag]
+alias = "pressure_closure_drag"
+value = "normalmode"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:968]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 10]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[updraft_number]
+alias = "updraft_number"
+value = 1
+type = "float"
+
+[pressure_closure_buoy]
+alias = "pressure_closure_buoy"
+value = "normalmode"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:963]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 11]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_quadrature_type]
+alias = "quadrature_type"
+value = "log-normal"
+type = "string"
+
+[updraft_number]
+alias = "updraft_number"
+value = 1
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:958]:
```suggestion
type = "float"
explanation = "Number of edmf updarfts."
```

Comment (trontrytel) [reply src/parameters.toml:958]: Shouldn't this be an integer?

File: src/parameters.toml [hunk 12]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_quadrature_order]
+alias = "quadrature_order"
+value = 3
+type = "float"
+
+[EDMF_thermodynamics_quadrature_type]
+alias = "quadrature_type"
+value = "log-normal"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:953]:
```suggestion
type = "string"
explanation = "Assumed distribution shape of the edmf state variables (q_tot and \theta_liq_ice) in the environment. Valid options are log-normal (default) and gaussian. Valid when running edmf with sgs quadrature choice"
```

File: src/parameters.toml [hunk 13]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_sgs]
+alias = "sgs"
+value = "mean"
+type = "string"
+
+[EDMF_thermodynamics_quadrature_order]
+alias = "quadrature_order"
+value = 3
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:948]:
```suggestion
type = "float"
explanation = "Number of numerical quadrature sampling points in one dimension. Valid when running edmf with sgs quadrature choice"
```

Comment (trontrytel) [reply src/parameters.toml:948]: This should be an integer.

File: src/parameters.toml [hunk 14]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[minimum_updraft_top]
+alias = "min_updraft_top"
+value = 500.0
+type = "float"
+
+[pressure_normalmode_buoy_coeff1]
+alias = "pressure_normalmode_buoy_coeff1"
+value = 0.12
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:908]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 15]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_normalmode_buoy_coeff2]
+alias = "pressure_normalmode_buoy_coeff2"
+value = 0.0
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:913]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 16]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_normalmode_adv_coeff]
+alias = "pressure_normalmode_adv_coeff"
+value = 0.1
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:918]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 17]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_normalmode_drag_coeff]
+alias = "pressure_normalmode_drag_coeff"
+value = 10.0
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:923]: @jiahe23 - do you remember what this one is?

File: src/parameters.toml [hunk 18]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_diagnostic_covar_limiter]
+alias = "diagnostic_covar_limiter"
+value = 0.001
+type = "float"
+
+[EDMF_thermodynamics_sgs]
+alias = "sgs"
+value = "mean"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:943]:
```suggestion
type = "string"
explanation = "Edmf environmental sub-grid scale model choice. Valid options are mean (default) and quadrature. The quadrature option switches on numerical sampling of 2-dimensional PDFs of edmf environmental state variables. Not documented in the literature (yet) - Jaruga et al in prep.""
```

File: src/parameters.toml [hunk 19]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_covariance_model]
+alias = "thermo_covariance_model"
+value = "diagnostic"
+type = "string"
+
+[EDMF_thermodynamics_diagnostic_covar_limiter]
+alias = "diagnostic_covar_limiter"
+value = 0.001
+type = "float"
```
Comment (trontrytel) [inline src/parameters.toml:938]:
```suggestion
type = "float"
explanation = "Regularization epsilon used in the denominator when computing covariances diagnostically."
```

File: src/parameters.toml [hunk 20]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[EDMF_thermodynamics_moisture_model]
+alias = "moisture_model"
+value = "equilibrium"
+type = "string"
+
+[EDMF_thermodynamics_covariance_model]
+alias = "thermo_covariance_model"
+value = "diagnostic"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:933]:
```suggestion
type = "string"
explanation = "Edmf covaraince model choice. Valid options are diagnostic (default) and prognostic."
```

File: src/parameters.toml [hunk 21]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[pressure_normalmode_drag_coeff]
+alias = "pressure_normalmode_drag_coeff"
+value = 10.0
+type = "float"
+
+[EDMF_thermodynamics_moisture_model]
+alias = "moisture_model"
+value = "equilibrium"
+type = "string"
```
Comment (trontrytel) [inline src/parameters.toml:928]:
```suggestion
type = "string"
explanation = " Edmf moisture model choice. The default (equilibrium) results in diagnosing cloud liquid and ice water content using saturation adjustment. An experimental nonequilibrium option introduces prognostic equations for cloud liquid and cloud ice.  Not documented in the literature (yet) - Benjamin et.al. in prep"
```

File: src/parameters.toml [hunk 22]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+alias = "smin_rm"
+value = 1.5
+type = "float"
+
+[mixing_length_l_max]
+alias = "l_max"
+value = 1.0e6
+type = "float"
+
+[entrainment_factor]
```
Comment (costachris) [inline src/parameters.toml:845]:
```suggestion
[entrainment_factor]
alias = "entrainment_factor"
value = 0.13
type = "float"
explanation = "Scaling constant for entrainment rate. See Cohen et al. (2022) Table 2 [https://doi.org/10.1029/2020MS002162]"
```

**Comment by trontrytel** [comment]
Also a comment for our future selves. Some of those parameters are from the standalone TurbulenceConvection code and are currently not supported by the ClimaAtmos version. I think they can be deleted?

File: src/parameters.toml [hunk 23]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+alias = "entrainment_factor"
+value = 0.13
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:848]:
```suggestion
type = "float"
explanation = "Scaling constant for entrainment rate. See Cohen et al. (2022) Table 2 [https://doi.org/10.1029/2020MS002162]"
```

File: src/parameters.toml [hunk 24]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_massflux_div_factor]
+alias = "entrainment_massflux_div_factor"
+value = 0.0
+type = "float"
+
+[turbulent_entrainment_factor]
+alias = "turbulent_entrainment_factor"
+value = 0.075
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:863]:
```suggestion
type = "float"
explanation = "Scaling constant for turbulent entrainment rate. See Cohen et al. (2022) Table 2 [https://doi.org/10.1029/2020MS002162]"
```

File: src/parameters.toml [hunk 25]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_factor]
+alias = "entrainment_factor"
+value = 0.13
+type = "float"
+
+[detrainment_factor]
+alias = "detrainment_factor"
+value = 0.51
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:853]:
```suggestion
type = "float"
explanation = "Scaling constant for detrainment rate. See Cohen et al. (2022) Table 2 & Eq 39, 40 [https://doi.org/10.1029/2020MS002162]"
```
```

File: src/parameters.toml [hunk 26]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[turbulent_entrainment_factor]
+alias = "turbulent_entrainment_factor"
+value = 0.075
+type = "float"
+
+[entrainment_smin_tke_coeff]
+alias = "entrainment_smin_tke_coeff"
+value = 0.3
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:868]:
```suggestion
type = "float"
explanation = "Scaling constant of TKE term in entrainment/detrainment inverse timescale.  See Cohen et al. (2022) Table 2 & eqn. 30 [https://doi.org/10.1029/2020MS002162]"
```

File: src/parameters.toml [hunk 27]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[updraft_mixing_frac]
+alias = "updraft_mixing_frac"
+value = 0.25
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:873]:
```suggestion
type = "float"
explanation = "Fraction of updraft air for buoyancy mixing in entrainment/detrainment formulation. See Cohen et al. (2022) Table 2 & eqn. 36 [https://doi.org/10.1029/2020MS002162]
```

File: src/parameters.toml [hunk 28]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_area_limiter_scale]
+alias = "area_limiter_scale"
+value = 10.0
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:878]:
```suggestion
type = "float"
explanation = "Constant scaling exponential argument in detrainment area limiter term. Parameter not described in the literature.
```

File: src/parameters.toml [hunk 29]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_area_limiter_power]
+alias = "area_limiter_power"
+value = 4.0
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:883]:
```suggestion
type = "float"
explanation = "Power of logistic term in detrainment area limiter term. Parameter not described in the literature.
```

File: src/parameters.toml [hunk 30]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_scale]
+alias = "entrainment_scale"
+value = 0.0004
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:888]:
```suggestion
type = "float"
explanation = "Dimensional constant scaling logistic function argument in entrainment/detrainment dry term. See Cohen et al. (2022) Table 2 & eqn. 37 [https://doi.org/10.1029/2020MS002162]
```

File: src/parameters.toml [hunk 31]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[entrainment_sorting_power]
+alias = "sorting_power"
+value = 2.0
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:893]:
```suggestion
type = "float"
explanation = "Sorting power for moisture term in entrainment/detrainment formulation. Controls the magnitude of the evaporative potential for a given relative humidity difference. See Cohen et al. (2022) Table 2 & eqn. 38 [https://doi.org/10.1029/2020MS002162]
```

File: src/parameters.toml [hunk 32]
```diff
@@ -763,3 +763,221 @@ alias = "avogad"
...
+
+[minimum_updraft_velocity]
+alias = "min_upd_velocity"
+value = 0.001
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:898]:
```suggestion
type = "float"
explanation = minimum updraft-environment vertical velocity difference allowable when computing inverse length scale in entrainment/detrainment formulation. Parameter not described in the literature.
```

File: src/parameters.toml (hunk 13 above)
Comment (nefrathenrici) [inline src/parameters.toml:948]:
```suggestion
type = "integer"
```

File: src/parameters.toml (hunk 11 above)
Comment (nefrathenrici) [reply src/parameters.toml:958]:
```suggestion
type = "integer"
```

File: src/parameters.toml [hunk 33]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+[EDMF_thermodynamics_quadrature_type]
+alias = "quadrature_type"
+value = "log-normal"
+type = "string"
+explanation = "Assumed distribution shape of the edmf state variables (q_tot and \theta_liq_ice) in the environment. Valid options are log-normal (default) and gaussian. Valid when running edmf with sgs quadrature choice. Not documented in the literature (yet) - Jaruga et al in prep.""
+
+[updraft_number]
+alias = "updraft_number"
+value = 1
+type = "float"
```
Comment (nefrathenrici) [inline src/parameters.toml:974]:
```suggestion
type = "integer"
```

File: src/parameters.toml [hunk 34]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+[mixing_length_Ri_crit]
+alias = "Ri_crit"
+value = 0.25
+type = "float"
+explanation = "Critical gradient Richardson number. It is an upper limit to the gradient Richardson number . See Li (2019) [https://doi.org/10.1016/j.atmosres.2018.09.015], Section 6.2 for details"
+
+[mixing_length_smin_ub]
+alias = "smin_ub"
+value = 0.1
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:833]:
```suggestion
type = "float"
explanation = "Lower limit for smooth minimum function in mixing length closure. See Lopez-Gomez et al. (2020) eqn. 40 [https://doi.org/10.1029/2020MS002161].
```

File: src/parameters.toml [hunk 35]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+
+[mixing_length_smin_rm]
+alias = "smin_rm"
+value = 1.5
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:838]:
```suggestion
type = "float"
explanation = "Upper ratio limit for smooth minimum function in mixing length closure. See Lopez-Gomez et al. (2020) eqn. 40 [https://doi.org/10.1029/2020MS002161].
```

File: src/parameters.toml [hunk 36]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+
+[mixing_length_l_max]
+alias = "l_max"
+value = 1.0e6
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:843]:
```suggestion
type = "float"
explanation = "Upper limit for length scale limiter in mixing length closure. See Lopez-Gomez et al. (2020) [https://doi.org/10.1029/2020MS002161].
```

File: src/parameters.toml [hunk 37]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+explanation = "Scaling constant for entrainment rate. See Cohen et al. (2022) Table 2 & Eq 39, 40 [https://doi.org/10.1029/2020MS002162]"
+
+[detrainment_factor]
+alias = "detrainment_factor"
+value = 0.51
+type = "float"
+explanation = "Scaling constant for detrainment rate. See Cohen et al. (2022) Table 2 & Eq 39, 40 [https://doi.org/10.1029/2020MS002162]"
+
+[entrainment_massflux_div_factor]
```
Comment (costachris) [inline src/parameters.toml:857]: This parameter was used for the plume test case in TC.jl, and has been removed as far as I can tell.

File: src/parameters.toml [hunk 38]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+
+[minimum_updraft_top]
+alias = "min_updraft_top"
+value = 500.0
+type = "float"
+
+[pressure_normalmode_buoy_coeff1]
+alias = "pressure_normalmode_buoy_coeff1"
+value = 0.12
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:918]:
```suggestion
type = "float"
explanation = "Factor multiplier for pressure buoyancy term, encapsulating virtual mass loading effect. See He et al. 2022 Eq 34 [https://doi.org/10.1002/essoar.10505084.2]
```

File: src/parameters.toml [hunk 39]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+
+[pressure_normalmode_buoy_coeff2]
+alias = "pressure_normalmode_buoy_coeff2"
+value = 0.0
+type = "float"
+
+[pressure_normalmode_adv_coeff]
+alias = "pressure_normalmode_adv_coeff"
+value = 0.1
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:928]:
```suggestion
type = "float"
explantation =  "Factor multiplier for pressure advection term (advection damping) in perturbation pressure closure. See He et al. 2022 Eq 34 [https://doi.org/10.1002/essoar.10505084.2]"
```

File: src/parameters.toml [hunk 40]
```diff
@@ -763,3 +763,241 @@ alias = "avogad"
...
+
+[pressure_normalmode_drag_coeff]
+alias = "pressure_normalmode_drag_coeff"
+value = 10.0
+type = "float"
```
Comment (costachris) [inline src/parameters.toml:933]:
```suggestion
type = "float"
explanation = "Factor multiplier for updraft pressure drag term in perturbation pressure closure. See He et al. 2022 Eq 34 [https://doi.org/10.1002/essoar.10505084.2]
```

File: src/parameters.toml (hunk 15 above)
Comment (costachris) [reply src/parameters.toml:923]: This doesn't appear to be read from the namelist and used in computations, either in TC.jl or ClimaAtmos

File: src/parameters.toml (hunk 10 above)
Comment (costachris) [reply src/parameters.toml:980]: This doesn't appear to be read from the namelist or used in computations (in neither TC.jl nor ClimaAtmos).

File: src/parameters.toml (hunk 9 above)
Comment (costachris) [reply src/parameters.toml:985]: This is read from the namelist [here](https://github.com/CliMA/ClimaAtmos.jl/blob/f8bb6756b77ce4838b14aa83a2f7de5d3233bba4/src/TurbulenceConvection/types.jl#L330) but never used in computations.

File: src/parameters.toml (hunk 14 above)
Comment (jiahe23) [reply src/parameters.toml:918]: Could you point me to the code where pressure is computed? I believe there are two parameters that controls the buoyancy related term but for simplification, we did not account for the effects from aspect ratio, that is, to calibrate the two parameters as a whole.

File: src/parameters.toml (hunk 15 above)
Comment (jiahe23) [reply src/parameters.toml:923]: @costachris  That is correct. coeff2 is always set to zero I believe?

File: src/parameters.toml (hunk 16 above)
Comment (jiahe23) [reply src/parameters.toml:928]: Yep, in Eq 34 as @costachris suggested below

File: src/parameters.toml (hunk 17 above)
Comment (jiahe23) [reply src/parameters.toml:933]: same

File: src/parameters.toml (hunk 9 above)
Comment (jiahe23) [reply src/parameters.toml:985]: This is a parameter from Scampy that allows a simple setup to test on the normal mode assumption but not used in our closure scheme. Feel free to remove it.

**Comment by odunbar** [comment]
Don't forget that if you go to ClimaParameters v0.6.0 you can look in the docstrings of the old ClimaParameters, which sometimes had some information on the parameter papers etc. e.g. in atmos.jl?

File: src/parameters.toml (hunk 14 above)
Comment (trontrytel) [reply src/parameters.toml:918]: I found it here:

I found it here: https://github.com/CliMA/ClimaAtmos.jl/blob/5b303e6e9304f03af0cf0cf74f1421bdf356892c/src/TurbulenceConvection/types.jl#L557-L563

which is later passed here:
https://github.com/CliMA/ClimaAtmos.jl/blob/5b303e6e9304f03af0cf0cf74f1421bdf356892c/src/TurbulenceConvection/types.jl#L172-L176

and used here:
https://github.com/CliMA/ClimaAtmos.jl/blob/5b303e6e9304f03af0cf0cf74f1421bdf356892c/src/TurbulenceConvection/closures/perturbation_pressure.jl#L65-L70

**Comment by trontrytel** [comment]
> Don't forget that if you go to ClimaParameters v0.6.0 you can look in the docstrings of the old ClimaParameters, which sometimes had some information on the parameter papers etc. e.g. in atmos.jl?

Maybe we can check later if we are consistent...

File: src/parameters.toml (hunk 10 above)
Comment (trontrytel) [reply src/parameters.toml:980]: 👍 Lets delete this

File: src/parameters.toml (hunk 9 above)
Comment (trontrytel) [reply src/parameters.toml:985]: Thanks for checking. 👍  Lets delete this

**Comment by trontrytel** [comment]
So to summarize `pressure_normalmode_buoy_coeff2`, `pressure_closure_buoy` and `pressure_closure_drag` can be deleted. I'll open a separate PR to ClimaAtmos to do so :)

File: src/parameters.toml (hunk 37 above)
Comment (nefrathenrici) [reply src/parameters.toml:857]: It's defined [here ](https://github.com/CliMA/ClimaAtmos.jl/blob/main/src/TurbulenceConvection/types.jl#L447)as ``c_div`` and looks like it's being used for [εδModelParams](https://github.com/CliMA/ClimaAtmos.jl/blob/main/src/TurbulenceConvection/types.jl#L515-L527).  Not sure if we should remove it.

**Comment by trontrytel** [comment]
Is this one ready to be merged after a squash?

File: src/parameters.toml (hunk 37 above)
Comment (trontrytel) [reply src/parameters.toml]: Deleted

**Pull request merged**